test/integration/graph/core/graph_manager_test.py

```diff
@@ -1194,9 +1194,9 @@ def test_visible_build_transitivity(self):
         # node, headers, lib, build, run
         _check_transitive(app, [(libd, True, True, False, False),
                                 (libc, True, True, False, False),
-                                (libb, False, False, True, False)])
+                                (libb, True, True, True, False)])
```

Comment:
This change sounds problematic. It means it is propagating headers and libs from the build context to the host context, which will be broken in majority of cases. I don't think that headers and libs that might be in a package containing an executable that we are using via ``tool-requires`` and some user decides to make it ``visible=True`` to propagate such executable should start receiving headers+libs of a potentially different architecture, as they will result in compilation and/or link errors. Note this trait means" the ``libb`` is contributing headers and libs to ``app``, which wouldn't be correct accross contexts.

Reply:
Yes that definitely makes sense. But shouldn't in that case `headers`, `libs`, `transitive_headers`, `transitive_libs` always be `False` in the build context?

Reply:
Should that maybe be enforced somewhere?